tests/crm/test_crm.py

```diff
@@ -0,0 +1,881 @@
+        configure_neighbors(amount=neighbours_num, interface=crm_interface[0], ip_ver=ip_ver, duthost=duthost,
+            test_name="test_crm_nexthop")
+
+        logger.info("Waiting {} seconds for SONiC to update resources...".format(SONIC_RES_UPDATE_TIME))
+        # Make sure SONIC configure expected entries
+        time.sleep(SONIC_RES_UPDATE_TIME)
+
+        # Get new "crm_stats_ipv[4/6]_nexthop" used and available counter value
+        new_crm_stats_nexthop_used, new_crm_stats_nexthop_available = get_crm_stats(get_nexthop_stats, duthost)
+
+    # Verify thresholds for "IPv[4/6] nexthop" CRM resource
+    verify_thresholds(duthost, crm_cli_res=RESTORE_CMDS["crm_cli_res"], crm_used=new_crm_stats_nexthop_used, crm_avail=new_crm_stats_nexthop_available)
+
+
+@pytest.mark.parametrize("ip_ver,neighbor", [("4", "2.2.2.2"), ("6", "2001::1")])
+def test_crm_neighbor(duthost, crm_interface, ip_ver, neighbor):
+    RESTORE_CMDS["crm_cli_res"] = "ipv{ip_ver} neighbor".format(ip_ver=ip_ver)
+    neighbor_add_cmd = "ip neigh replace {neighbor} lladdr 11:22:33:44:55:66 dev {iface}".format(neighbor=neighbor, iface=crm_interface[0])
+    neighbor_del_cmd = "ip neigh del {neighbor} lladdr 11:22:33:44:55:66 dev {iface}".format(neighbor=neighbor, iface=crm_interface[0])
+
+    # Get "crm_stats_ipv[4/6]_neighbor" used and available counter value
+    get_neighbor_stats = "redis-cli --raw -n 2 HMGET CRM:STATS crm_stats_ipv{ip_ver}_neighbor_used crm_stats_ipv{ip_ver}_neighbor_available".format(ip_ver=ip_ver)
+    crm_stats_neighbor_used, crm_stats_neighbor_available = get_crm_stats(get_neighbor_stats, duthost)
+
+    # Add neighbor
+    RESTORE_CMDS["test_crm_neighbor"].append(neighbor_del_cmd)
+    duthost.command(neighbor_add_cmd)
+
+    # Make sure CRM counters updated
+    time.sleep(CRM_UPDATE_TIME)
+
+    # Get new "crm_stats_ipv[4/6]_neighbor" used and available counter value
+    new_crm_stats_neighbor_used, new_crm_stats_neighbor_available = get_crm_stats(get_neighbor_stats, duthost)
+
+    # Verify "crm_stats_ipv4_neighbor_used" counter was incremented
+    pytest_assert(new_crm_stats_neighbor_used - crm_stats_neighbor_used >= 1, \
+        "\"crm_stats_ipv4_neighbor_used\" counter was not incremented")
+    # Verify "crm_stats_ipv4_neighbor_available" counter was decremented
+    pytest_assert(crm_stats_neighbor_available - new_crm_stats_neighbor_available >= 1, \
+        "\"crm_stats_ipv4_neighbor_available\" counter was not decremented")
+
+    # Remove neighbor
+    duthost.command(neighbor_del_cmd)
+
+    # Make sure CRM counters updated
+    time.sleep(CRM_UPDATE_TIME)
+
+    # Get new "crm_stats_ipv[4/6]_neighbor" used and available counter value
+    new_crm_stats_neighbor_used, new_crm_stats_neighbor_available = get_crm_stats(get_neighbor_stats, duthost)
+
+    # Verify "crm_stats_ipv4_neighbor_used" counter was decremented
+    pytest_assert(new_crm_stats_neighbor_used - crm_stats_neighbor_used >= 0, \
```

Comment:
Why ```used``` value is compared like ```>= 0```? I would expect it to be equal to ```0```, since we add just on neighbor.